.github/workflows/gen_decorators.js

```diff
@@ -0,0 +1,105 @@
+/// generate jsb decorators 
+
+
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-var-requires */
+const path = require('path');
+const fs = require('fs');
+const os = require('os');
+
+const contentLines = [];
+process.on('message', (msg) => {
+    if (msg.event === 'write-decorator') {
+        contentLines.push({ className: msg.className, lines: msg.lines });
+    }
+});
+
+const engineDir = path.resolve(__dirname, '../..');
+const tmpDir = os.tmpdir();
+
+const BUILD_CONFIG = {
+    options: {
+        engine: engineDir,
+        platform: 'HTML5',
+        moduleFormat: 2,
+        compress: false,
+        split: false,
+        ammoJsWasm: 'fallback',
+        assetURLFormat: 'runtime-resolved',
+        noDeprecatedFeatures: false,
+        engineName: 'cocos-js',
+        sourceMaps: false,
+        features: [
+            '2d', '3d',
+            'animation', 'audio',
+            'base', 'dragon-bones',
+            'gfx-webgl', 'gfx-webgl2',
+            'intersection-2d', 'light-probe',
+            'marionette', 'particle',
+            'particle-2d', 'physics-2d-box2d',
+            'physics-ammo', 'primitive',
+            'profiler', 'skeletal-animation',
+            'spine', 'terrain',
+            'tiled-map', 'tween',
+            'ui', 'video',
+            'websocket', 'webview',
+        ],
```

Comment:
Great idea! However, there is still an issue that needs to be addressed, which is the potential conflicts and dependencies between modules. Dependencies can be managed easily, but what about the case where multiple backends, such as physical modules, need to be selected? It seems like we might need to hardcode the solution for this scenario.

Reply:
I suggest that hardcoding the solution for the current scenario is a feasible approach. In case of a missing module, it should be relatively easy to locate and modify the configuration.

Reply:
got it